### PR TITLE
Fix for FortiSwitch RPM/percentage fans

### PR DESCRIPTION
**Pull request by rudybroersma**
This is a fix for:
https://github.com/librenms/librenms/issues/15730

*all* FortiSwitches fan speed sensors have entPhySensorType = RPM, but the entPhySensorValue is returned as a percentage value. I've reached out to FortiNet about this issue. The conclusion was they are not going to fix it any time soon and not with current hardware.

Apparantly this is a hardware limitation. The actual sensors only report a percentage value and not an RPM value.
ENTITY-SENSOR-MIB does not have a 'percentage' as valid entPhySensorType. FortiNet was unwilling to change this to 'other'.

Verified this issue on 100-series, 200-series, 400-series and 1000-series. All have percentage values for RPM fan speeds.

So.. this patch checks for a entPhySensorType of RPM and os FortiSwitch. If both are true, we change sensor type to percent. It also corrects the divisor and the sensor value used at time of discovery. 

```
$ snmpwalk X entPhySensorType
ENTITY-SENSOR-MIB::entPhySensorType.3 = INTEGER: celsius(8)
ENTITY-SENSOR-MIB::entPhySensorType.4 = INTEGER: rpm(10)
$ snmpwalk X entPhySensorValue
ENTITY-SENSOR-MIB::entPhySensorValue.3 = INTEGER: 38
ENTITY-SENSOR-MIB::entPhySensorValue.4 = INTEGER: 50
```

The last line here with value 50 is a percentage value, not a RPM value.

And a small fix for the FortiSwitch discovery YAML. It had a wrong 'index' (fgHw is for FortiGate, not FortiSwitch) and the discovery value was wrong. It should be OperStatus instead of SensorValue. Another FortiNet quirk.

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by rudybroersma** [comment]
I'd like to add that FortiSwitch has an additional bug.. Their entPhysensorPrecision is incorrectly set to 1. Should be set to 0.
This patch also corrects that.

I have a ticket open at FortiNet regarding the precision value in ticket 9216884

File: includes/definitions/discovery/fortiswitch.yaml [hunk 1]
```diff
@@ -32,9 +32,9 @@ modules:
                 -
                     oid: entPhySensorTable
                     num_oid: '.1.3.6.1.2.1.99.1.1.1.5.{{ $index }}'
-                    index: 'fgHwSensorEntIndex.{{ $index }}'
```
Comment (laf) [inline includes/definitions/discovery/fortiswitch.yaml:35]: Are these changes related to the fan issue still?

**Comment by devience** [comment]
I suppose the multiplier is wrong on the temperature as well, since it shows absurd temperatures like 4.1 degrees Celsius:

Screenshot_1

Any chance to fix two birds with one stone?

**Comment by rudybroersma** [comment]
> I suppose the multiplier is wrong on the temperature as well, since it shows absurd temperatures like 4.1 degrees Celsius:
> Any chance to fix two birds with one stone?

Yeah, that's already implemented. 
https://github.com/librenms/librenms/pull/15829/files#r1502528303

File: includes/discovery/sensors/entity-sensor.inc.php [hunk 2]
```diff
@@ -156,6 +156,14 @@
                 }
                 $descr = preg_replace('/[T|t]emperature[|s]/', '', $descr);
             }
+
+            // Fix for FortiSwitch - ALL FortiSwitches as of 14/2/2024 output fan speeds as percentages while entPhySensorType is RPM.
+            if ($device['os'] == 'fortiswitch' && $entry['entPhySensorType'] == 'rpm') {
+                $type = 'percent';
+                $divisor = 1;
```
[the remark this answers is not in the document]
Comment (rudybroersma) [reply includes/discovery/sensors/entity-sensor.inc.php:163]: I have an open FortiNet ticket regarding their incorrect PrecisionValue. They have it set to 1 where it should be 0.

**Review by Jellyfrog — APPROVED** [review]

**Comment by rudybroersma** [comment]
> > I suppose the multiplier is wrong on the temperature as well, since it shows absurd temperatures like 4.1 degrees Celsius:
> > Any chance to fix two birds with one stone?
> 
> I have an open ticket regarding their incorrect precision value. I'd like to hold off from patching it in Libre before I get an answer from them

Got feedback from Fortinet today. They created a single Mantis bug report for all my 4 reported SNMP bugs in their FortiSwitch product.

Mantis bug ID 1008119
Fortinet Support ticket ID 9216884

**Comment by rudybroersma** [comment]
> I suppose the multiplier is wrong on the temperature as well, since it shows absurd temperatures like 4.1 degrees Celsius:
> 
> Any chance to fix two birds with one stone?

They fixed this after my bug report in FortiSwitchOS 7.4.3.
